packages/e2e-tests/specs/editor/various/autocomplete-and-mentions.test.js

```diff
@@ -0,0 +1,410 @@
+			`should not insert disabled ${ type }s`,
+			isNotMention,
+			async () => {
+				await clickBlockAppender();
+				// The 'Grapes' option is disabled in our test plugin, so it should not insert the grapes emoji
+				await page.keyboard.type( 'Sorry, we are all out of ~g' );
+				await page.waitForXPath(
+					'//button[@role="option"][text()="🍇 Grapes"]'
+				);
+				await page.keyboard.press( 'Enter' );
+				await page.keyboard.type( ' grapes.' );
+				// This `expect` is in wrapped in the `itif` helper's callback, so it will trip the linter
+				// eslint-disable-next-line jest/no-standalone-expect
+				expect( await getEditedPostContent() ).toMatchInlineSnapshot( `
+						"<!-- wp:paragraph -->
+						<p>Sorry, we are all out of ~g grapes.</p>
+						<!-- /wp:paragraph -->"
+						` );
+			}
+		);
+
+		it( 'should allow newlines after multiple completions', async () => {
+			const testData = {};
+			if ( type === 'mention' ) {
+				testData.triggerString = '@bu';
+				testData.optionPath = '//*[contains(text(),"Buddy Elf")]';
+				testData.snapshot = `
+					"<!-- wp:paragraph -->
+					<p>@buddytheelf test</p>
+					<!-- /wp:paragraph -->
+
+					<!-- wp:paragraph -->
+					<p>@buddytheelf test</p>
+					<!-- /wp:paragraph -->
+
+					<!-- wp:paragraph -->
+					<p>@buddytheelf test</p>
+					<!-- /wp:paragraph -->
+
+					<!-- wp:paragraph -->
+					<p>@buddytheelf test</p>
+					<!-- /wp:paragraph -->
+
+					<!-- wp:paragraph -->
+					<p></p>
+					<!-- /wp:paragraph -->"
+					`;
+			} else if ( type === 'option' ) {
+				testData.triggerString = '~b';
+				testData.optionPath = '[text()="🫐 Blueberry"]';
+				testData.snapshot = `
+					"<!-- wp:paragraph -->
+					<p>🫐 test</p>
+					<!-- /wp:paragraph -->
+
+					<!-- wp:paragraph -->
+					<p>🫐 test</p>
+					<!-- /wp:paragraph -->
+
+					<!-- wp:paragraph -->
+					<p>🫐 test</p>
+					<!-- /wp:paragraph -->
+
+					<!-- wp:paragraph -->
+					<p>🫐 test</p>
+					<!-- /wp:paragraph -->
+
+					<!-- wp:paragraph -->
+					<p></p>
+					<!-- /wp:paragraph -->"
+					`;
+			}
+
+			await clickBlockAppender();
+
+			for ( let i = 0; i < 4; i++ ) {
+				await page.keyboard.type( testData.triggerString );
+				await page.waitForXPath(
+					`//button[@role="option"]${ testData.optionPath }`
+				);
+				await page.keyboard.press( 'Enter' );
+				await page.keyboard.type( ' test' );
+				await page.keyboard.press( 'Enter' );
+			}
+
+			expect( await getEditedPostContent() ).toMatchInlineSnapshot(
+				testData.snapshot
+			);
+		} );
+	} );
+
+	// The following test attempts to cover an infinite loop regression (https://github.com/WordPress/gutenberg/issues/41709).
+	// Unfortunately, the regression (if present) crashes the tests, as well as the editor,
+	// so it's skipped for now in the hopes we can find a way around that in the future.
+	// eslint-disable-next-line jest/no-disabled-tests
+	it.skip( 'should insert elements from multiple completers in a single block', async () => {
```

Comment:
Without the changes from https://github.com/WordPress/gutenberg/pull/41382, it looks like this test is passing:

![Screenshot 2022-09-01 at 10 36 29](https://user-images.githubusercontent.com/1083581/187870567-510da090-32cd-4295-8b0b-f4a902197885.png)

It would be a pity not being able to enable this test. Have you tried detecting errors with `page.on('error' ...)` or `page.on('pageerror' ...)` ? Or surrounding some code in this test with a `try/catch` ? This [link](https://github.com/puppeteer/puppeteer/issues/3709) may be of help

Reply:
Agreed, I'd love to be able to test this! So far, I haven't had any luck with `page.on('error')`, `page.on('pageerror')`, `page.on('crash')`, `try/catch` or (after your link above) a `process.on('unhandledRejection',...)` with no luck 😢 

@WunderBart I know you're good with e2e puzzles, does any magic for this kind of stuck test come to mind?

Reply:
Sorry for the delayed response! I'll look into it now. One question for now, though - any reason we're using Puppeteer for this one? I'm asking because we've been migrating all the E2Es to Playwright, so it would make more sense to me to use Playwright for a new test. IIRC, this is where the migrated tests currently are: [/test/e2e/specs/editor/various](https://github.com/WordPress/gutenberg/tree/1c3bc0ac022310bd0013dd572ba0368c80374806/test/e2e/specs/editor/various)

Reply:
> One question for now, though - any reason we're using Puppeteer for this one? I'm asking because we've been migrating all the E2Es to Playwright, so it would make more sense to me to use Playwright for a new test. IIRC, this is where the migrated tests currently are: [/test/e2e/specs/editor/various](https://github.com/WordPress/gutenberg/tree/1c3bc0ac022310bd0013dd572ba0368c80374806/test/e2e/specs/editor/various)

Good catch, no specific reason - I wasn't thinking about Playwright at the time and just created the new tests in the same directory as the old user mentions tests.

I've just been looking into moving the new tests in this PR over to Playwright, and hit a bit of a snag. 

Unless I'm misreading things, it looks like some of the utils these new tests rely on haven't yet been migrated over to their Playwright counterparts.  Creating these tests there would mean first migrating (or rewriting tests to avoid) the following utils:
- `activatePlugin`
- `deactivatePlugin`
- `createUser`
- `deleteUser`
- `clickBlockAppender`
- `getEditedPostContent`

As each of those would be a separate PR holding this one up, I'm wondering if it doesn't make more sense to merge the new tests as-is, and migrate them later, rather than hold this one up while the above is sorted out.

Curious what you both think @ciampo @WunderBart

Reply:
- [`activatePlugin`](https://github.com/WordPress/gutenberg/blob/f62f2a8d0d72aa29d1368883bd566b6444940b03/packages/e2e-test-utils-playwright/src/request-utils/plugins.ts#L35-L54)
- [`deactivatePlugin`](https://github.com/WordPress/gutenberg/blob/f62f2a8d0d72aa29d1368883bd566b6444940b03/packages/e2e-test-utils-playwright/src/request-utils/plugins.ts#L56-L75)
- [`createUser`](https://github.com/WordPress/gutenberg/blob/f62f2a8d0d72aa29d1368883bd566b6444940b03/packages/e2e-test-utils-playwright/src/request-utils/users.ts#L47-L83)
- [`deleteUser`](https://github.com/WordPress/gutenberg/blob/f62f2a8d0d72aa29d1368883bd566b6444940b03/packages/e2e-test-utils-playwright/src/request-utils/users.ts#L85-L100)
- [`getEditedPostContent`](https://github.com/WordPress/gutenberg/blob/f62f2a8d0d72aa29d1368883bd566b6444940b03/packages/e2e-test-utils-playwright/src/editor/get-edited-post-content.ts#L6-L18)

are available in the `e2e-test-utils-playwright` package. We found the `clickBlockAppender` util less helpful and can often be done with a simple click on a role selector (`page.click( 'role=button[name="Add default block"i]' )`).

The recommendation is to write new tests with Playwright, but I agree it has a learning curve (and lacks some documentation for `e2e-test-utils-playwright`) so no worries if you want to write it in Puppeteer first.

However, I think it'll be easier to just start with Playwright since it encourages better practices for writing resilient e2e tests. Migrating tests from Puppeteer works too but often requires shifting the mind around to adapt to a different kind of thinking.

Reply:
@chad1008 would you be up for writing these tests in Playwright? It would be a good way to get started with the testing framework anyway

Reply:
Updated with my first pass at a Playwright refactor. All tests (including the infinite loop regression) are active and passing locally, but this is my first foray into Playwright so I'm sure there are things I can improve.